sno/crs_util.py

```diff
@@ -83,49 +83,81 @@ def parse_authority(crs):
     return spatial_ref.GetAuthorityName(None), spatial_ref.GetAuthorityCode(None)
 
 
-def get_identifier_str(crs, authority=None):
+def get_identifier_str(crs):
     """
-    Given a CRS, generate a stable, unique identifier for it of type 'str'. Eg: "EPSG:2193"
+    Given a CRS, find or generate a stable, unique identifier for it of type 'str'.
+    Eg: "EPSG:2193" or "CUSTOM:201234"
     """
-    if authority is not None:
-        auth_name, auth_code = authority
-    else:
-        auth_name, auth_code = parse_authority(crs)
+    result = _find_identifier_str(crs)
+    if not result:
+        result = f"CUSTOM:{_generate_identifier_int(crs)}"
+    return result.replace("/", "_")
+
 
+def _find_identifier_str(crs):
+    """Given a CRS, find a sensible identifier string for it from within the WKT."""
+    auth_name, auth_code = parse_authority(crs)
+    # Use AUTH_NAME:AUTH_CODE if both are set:
     if auth_name and auth_code:
         return f"{auth_name}:{auth_code}"
+    # Use AUTH_NAME or AUTH_CODE if one of them is set and probably a real identifier:
     code = auth_name or auth_code
-    if code and code.strip() not in ("0", "EPSG"):
+    if code and code.strip() not in ("", "0", "EPSG", "ESRI"):
```

Comment:
http://www.geopackage.org/spec/#r11 suggests -1 is a "normal" value for "i don't have a CRS" too, depending on the source. 

But that's unlikely to come out of WKT/GDAL, more likely to see in a GPKG or PostGIS  DB.